sonic_platform_base/sonic_sfp/sff8472.py

```diff
@@ -828,49 +821,23 @@ def calc_rx_power(self, eeprom_data, offset, size):
-
-                off = self.dom_ext_calibration_constants['RX_PWR_3']['offset']
-                rx_pwr_byte3 = int(eeprom_data[off], 16)
-                rx_pwr_byte2 = int(eeprom_data[off + 1], 16)
-                rx_pwr_byte1 = int(eeprom_data[off + 2], 16)
-                rx_pwr_byte0 = int(eeprom_data[off + 3], 16)
-                rx_pwr_3 = (rx_pwr_byte3 << 24) | (rx_pwr_byte2 << 16) | (rx_pwr_byte1 << 8) | (rx_pwr_byte0 & 0xff)
-
-                off = self.dom_ext_calibration_constants['RX_PWR_2']['offset']
-                rx_pwr_byte3 = int(eeprom_data[off], 16)
-                rx_pwr_byte2 = int(eeprom_data[off + 1], 16)
-                rx_pwr_byte1 = int(eeprom_data[off + 2], 16)
-                rx_pwr_byte0 = int(eeprom_data[off + 3], 16)
-                rx_pwr_2 = (rx_pwr_byte3 << 24) | (rx_pwr_byte2 << 16) | (rx_pwr_byte1 << 8) | (rx_pwr_byte0 & 0xff)
-
-                off = self.dom_ext_calibration_constants['RX_PWR_1']['offset']
-                rx_pwr_byte3 = int(eeprom_data[off], 16)
-                rx_pwr_byte2 = int(eeprom_data[off + 1], 16)
-                rx_pwr_byte1 = int(eeprom_data[off + 2], 16)
-                rx_pwr_byte0 = int(eeprom_data[off + 3], 16)
-                rx_pwr_1 = (rx_pwr_byte3 << 24) | (rx_pwr_byte2 << 16) | (rx_pwr_byte1 << 8) | (rx_pwr_byte0 & 0xff)
-
-                off = self.dom_ext_calibration_constants['RX_PWR_0']['offset']
-                rx_pwr_byte3 = int(eeprom_data[off], 16)
-                rx_pwr_byte2 = int(eeprom_data[off + 1], 16)
-                rx_pwr_byte1 = int(eeprom_data[off + 2], 16)
-                rx_pwr_byte0 = int(eeprom_data[off + 3], 16)
-                rx_pwr_0 = (rx_pwr_byte3 << 24) | (rx_pwr_byte2 << 16) | (rx_pwr_byte1 << 8) | (rx_pwr_byte0 & 0xff)
-
-                rx_pwr = (rx_pwr_4 * result) + (rx_pwr_3 * result) + (rx_pwr_2 * result) + (rx_pwr_1 * result) + rx_pwr_0
-
-                result = float(result * 0.0001)
+                # SFF-8472 Rev 12.4 section 9.3 5)
+                # Rx_PWR (uW) = Rx_PWR(4) * Rx_PWR_ADe4 (16 bit unsigned integer) +
+                #               Rx_PWR(3) * Rx_PWR_ADe3 (16 bit unsigned integer) +
+                #               Rx_PWR(2) * Rx_PWR_ADe2 (16 bit unsigned integer) +
+                #               Rx_PWR(1) * Rx_PWR_AD (16 bit unsigned integer) +
+                #               Rx_PWR(0)
+                rx_pwr_ad = result
+                result = 0
+                for i in range(0, 5):
+                    field_name = f'RX_PWR_{i}'
+                    rx_pwr = self.dom_ext_calibration_constants[field_name]
+                    off = rx_pwr['offset']
+                    size = rx_pwr['size']
+                    coeff = self.float_from_bytes(eeprom_data[off:off+size])
+                    result += coeff * rx_pwr_ad ** i
+
+                result = result * 0.0001 # uW to mW
```

Comment:
Here the comment should make it clear that the `result` value is in 0.1 uW units, not uW (the formula in the documentation at section 9.3 is _also_ misleading in that regards). Otherwise the conversion factor may be interpreted off by an order of magnitude by an unsuspecting reader.